classes/ActionScheduler_WPCLI_QueueRunner.php

```diff
@@ -0,0 +1,223 @@
+ */
+class ActionScheduler_WPCLI_QueueRunner extends ActionScheduler_Abstract_QueueCleaner {
+
+	/** @var array */
+	protected $actions;
+
+	/** @var  ActionScheduler_ActionClaim */
+	protected $claim;
+
+	/** @var ActionScheduler_QueueCleaner */
+	protected $cleaner;
+
+	/** @var string */
+	protected $id;
+
+	/** @var ActionScheduler_FatalErrorMonitor */
+	protected $monitor;
+
+	/** @var \cli\progress\Bar */
+	protected $progress_bar;
+
+	/**
+	 * ActionScheduler_WPCLI_QueueRunner constructor.
+	 *
+	 * @param ActionScheduler_Store             $store
+	 * @param ActionScheduler_FatalErrorMonitor $monitor
+	 * @param ActionScheduler_QueueCleaner      $cleaner
+	 *
+	 * @throws Exception When this is not run within WP CLI
+	 */
+	public function __construct(
+		ActionScheduler_Store $store,
+		ActionScheduler_FatalErrorMonitor $monitor,
+		ActionScheduler_QueueCleaner $cleaner
+	) {
+		if ( ! ( defined( 'WP_CLI' ) && WP_CLI ) ) {
+			throw new Exception( __( 'The ' . __CLASS__ . ' class can only be run within WP CLI.', 'action-scheduler' ) );
+		}
+
+		$this->store   = $store;
+		$this->monitor = $monitor;
+		$this->cleaner = $cleaner;
+	}
+
+	/**
+	 * Set up the Queue before processing.
+	 *
+	 * @author Jeremy Pry
+	 *
+	 * @param int  $batch_size The batch size to process.
+	 * @param bool $force      Whether to force running even with too many concurrent processes.
+	 *
+	 * @return int The number of actions that will be run.
+	 */
+	public function setup( $batch_size, $force = false ) {
+		$this->run_cleanup();
+		$this->add_hooks();
+
+		// Check to make sure there aren't too many concurrent processes running.
+		$claim_count = $this->store->get_claim_count();
+		$too_many    = $claim_count >= apply_filters( 'action_scheduler_queue_runner_concurrent_batches', 5 );
+		if ( $too_many ) {
+			if ( $force ) {
+				WP_CLI::warning( __( 'There are too many concurrent batches, but the run is forced to continue.', 'action-scheduler' ) );
+			} else {
+				WP_CLI::error( __( 'There are too many concurrent batches.', 'action-scheduler' ) );
+			}
+		}
+
+		// Stake a claim and store it.
+		$this->claim = $this->store->stake_claim( $batch_size );
+		$this->monitor->attach( $this->claim );
+		$this->actions = $this->claim->get_actions();
+		$this->id      = $this->claim->get_id();
+
+		return count( $this->actions );
+	}
+
+	/**
+	 * Add our hooks to the appropriate actions.
+	 *
+	 * @author Jeremy Pry
+	 */
+	protected function add_hooks() {
+		add_action( 'action_scheduler_before_execute', array( $this, 'before_execute' ) );
+		add_action( 'action_scheduler_after_execute', array( $this, 'after_execute' ) );
+		add_action( 'action_scheduler_failed_execution', array( $this, 'action_failed' ) );
+	}
+
+	/**
+	 * Set up the WP CLI progress bar.
+	 *
+	 * @author Jeremy Pry
+	 */
+	protected function setup_progress_bar() {
+		$count              = count( $this->actions );
+		$this->progress_bar = \WP_CLI\Utils\make_progress_bar(
```

Comment:
Yes, the file relies on the existing autoloading.

I didn't do a `use` statement because it was a single function call. Plus, it is a function instead of a class, so even with `use WP_CLI\Utils;`, you would still have to call `Utils\make_progress_bar()`.